envoy/datadog_checks/envoy/envoy.py

```diff
@@ -69,9 +72,14 @@ def check(self, instance):
             try:
                 metric, tags, method = parse_metric(envoy_metric)
             except UnknownMetric:
-                if envoy_metric not in self.unknown_metrics:
-                    self.log.debug('Unknown metric `{}`'.format(envoy_metric))
-                    self.unknown_metrics.add(envoy_metric)
+                self.unknown_metrics[envoy_metric] += 1
+                self.log.debug('Unknown metric `{}`'.format(envoy_metric))
+                continue
+            except UnknownTags as e:
+                unknown_tags = str(e).split('|||')
+                for tag in unknown_tags:
+                    self.unknown_tags[tag] += 1
```

Comment:
I'm concerned that both of these objects could get fairly large. (Keep in mind that the agent could be running and collecting stats from envoy for weeks or months without restarting)

Reply:
This will have about the same footprint as before actually due to the fact that I've split off some of the errors `self.unknown_metrics` would get into `self.unknown_tags` for better debugging. In practice, this would be equally as unlikely to occur because Envoy introducing a new tag option would actually in most cases involve adding a new metric.

See in https://www.envoyproxy.io/docs/envoy/latest/configuration/http_filters/ip_tagging_filter#statistics how the `ip_tagging.` namespace can have 0 or 1 tag.

The only other time this would occur would be if a `.` delimited tag matches an upcoming metric part. Since our parser is greedy, this is the same behavior as Envoy's regular expressions.

Reply:
Oh yes sorry not for user debugging but rather for testing/development. We can pass in a mocked `/stats` output and see what metrics/tags are causing the most errors. Alternatively, we could keep the data in one dict.

Reply:
I see the utility as a development tool, but it seems like it could be enabled through a silent option in the config (off by default) rather than having it potentially impact users. I think this might be out of scope for this PR through so I can approve it and it can be resolved later

Reply:
(Even a small amount of memory being used by this could be potentially frustrating for users)